gensim/models/word2vec.py

```diff
@@ -1235,6 +1235,9 @@ def most_similar(self, positive=[], negative=[], topn=10, restrict_vocab=None):
             raise ValueError("cannot compute similarity with no input")
         mean = matutils.unitvec(array(mean).mean(axis=0)).astype(REAL)
 
+        if indexer is not None:
+            return indexer.most_similar(mean, topn)
+
```

Comment:
Just a thought for potential future refactorings: this callback is suggestive that the lookup/positive-negative/mean vector-combination steps may deserve being factored out to its own method, say `combo()`. 

Then instead of `model.most_similar(positive, negative)`, a more-explicit idiom would be `model.most_similar(model.combo(positive, negative))`. And rather than `model.most_similar(positive, negative, indexer=indexer)` it'd be `indexer.most_similar(model.combo(positive, negative)`.